src/main/java/com/bookingApp/bookingApp/service/ReservationService.java

```diff
@@ -0,0 +1,181 @@
+import com.bookingApp.bookingApp.model.Screening;
+import com.bookingApp.bookingApp.model.SeatScreening;
+import com.bookingApp.bookingApp.model.Ticket;
+import com.bookingApp.bookingApp.repository.ReservationRepository;
+import com.bookingApp.bookingApp.repository.ScreeningRepository;
+import com.bookingApp.bookingApp.repository.SeatScreeningRepository;
+import com.bookingApp.bookingApp.repository.TicketRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+@Service
+public class ReservationService implements IReservationService {
+    private final ScreeningRepository screeningRepository;
+    private final SeatScreeningRepository seatScreeningRepository;
+    private final ReservationRepository reservationRepository;
+    private final TicketRepository ticketRepository;
+
+    @Autowired
+    public ReservationService(ScreeningRepository screeningRepository, SeatScreeningRepository seatScreeningRepository,
+                                ReservationRepository reservationRepository, TicketRepository ticketRepository){
+        this.screeningRepository = screeningRepository;
+        this.seatScreeningRepository = seatScreeningRepository;
+        this.reservationRepository = reservationRepository;
+        this.ticketRepository = ticketRepository;
+    }
+    public List<ScreeningsDto> findAll(){
+        List<ScreeningsDto> screeningsDtos = new ArrayList<>();
+        List<Screening> screenings = screeningRepository.findAll();
+        screenings.forEach(screening -> {
+            ScreeningsDto s = new ScreeningsDto();
+            s.setScreeningId(screening.getId());
+            s.setDate(screening.getScreeningDate());
+            s.setTitle(screening.getMovie().getTitle());
+            s.setRoom(screening.getSeatScreenings().get(0).getSeat().getRoom().getRoomNumber());
+
+            screeningsDtos.add(s);
+        });
+
+        Collections.sort(screeningsDtos, new Comparator<ScreeningsDto>() {
+            @Override
+            public int compare(ScreeningsDto o1, ScreeningsDto o2) {
+                if(o1.getTitle().compareTo(o2.getTitle())==0){
+                    return o1.getDate().compareTo(o2.getDate());
+                }else {
+                    return o1.getTitle().compareTo(o2.getTitle());
+                }
+            }
+        });
+
+        return screeningsDtos;
+    }
+    @Override
+    public List<ScreeningsDto> findAll(LocalDateTime dateFrom, LocalDateTime dateTo) {
+        List<ScreeningsDto> screeningsDtos = new ArrayList<>();
+        List<Screening> screenings = screeningRepository.readScreeningsByScreeningDateBetween(dateFrom,dateTo);
+        screenings.forEach(screening -> {
```

Comment:
czy ten kod nie jest zdublowany?

Reply:
jest, powinienem to wydzielic do osobnej metody